src/Simple/JSON.purs

```diff
@@ -96,3 +103,33 @@ instance readFieldsNil ::
   ) => ReadForeignFields Nil row where
   getFields _ _ _ =
     pure $ to {}
+
+
+-- | The purpose of the `ReadKey` class is to turn strings into map keys,
+-- | so we can read generic `Map`s from `StrMap`s.
+class ReadKey a where
+  readKey :: String -> F a
```

Comment:
My broken attempt at removing the type class is here:

https://github.com/jacereda/purescript-simple-json/commit/87e3e58611fd9e2ef1e86ab9f04dfb1bd25e307c